### PR TITLE
Makes the TRAIT_STUNRESISTANCE trait actually worth a damn

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

The TRAIT_STUNRESISTANCE trait, in addition to its other (mostly baton-related) effects, now halves the stamina damage you receive from stun batons.

## Why It's Good For The Game

Currently, the TRAIT_STUNRESISTANCE trait doesn't do its job of being a stun baton resistance trait/counter very well, as even though it reduces the knockdown from stun batons, it doesn't reduce the stamina damage from stun batons, causing you to still be 2HKO'd by them and still be slowed down (from (stamina) damage slowdown) by the first stun baton strike you receive.

This PR fixes that issue, making the TRAIT_STUNRESISTANCE trait (and, in turn, the pump-up reagent) the baton counter it was meant to be.

## Changelog
:cl: ATHATH
balance: The TRAIT_STUNRESISTANCE trait, in addition to its other (mostly baton-related) effects, now halves the stamina damage you receive from stun batons. In layman's terms, pump-up got buffed to actually give you some half-decent protection against stun batons.
/:cl:

### CODE REVIEW

**Comment by Akrilla** [comment]
good change

**Comment by pireamaineach** [comment]
Surprisingly, removing stun resistance was probably the only good thing Anne's P.R. does.

I think as a trait it's just inherently a bad idea. Anything that could drastically affect combat while being impossible to predict is unfun. Buffing this, especially, seems really lame to me.

**Comment by spookydonut** [comment]
make a new trait dont shoehorn more shit onto a well defined trait

**Comment by Rohesie** [comment]
Either this trait describes what it's named after or it's not doing its job.
If you want a stamina damage modifier add one and make it customizable, don't add yet another snowflake check with a static effect.

**Comment by necromanceranne** [comment]
Could you rename it to baton resistance since it's only for batons?

**Comment by ATH1909** [comment]
> Could you rename it to baton resistance since it's only for batons?

I would, but then I'd have to do something other than make webedits.

**Comment by ATH1909** [comment]
> make a new trait dont shoehorn more shit onto a well defined trait

literally all stun resistance does is make stun batons (not even non-stun batons) knock you down for less time

**Comment by ATH1909** [comment]
> Either this trait describes what it's named after or it's not doing its job.
> If you want a stamina damage modifier add one and make it customizable, don't add yet another snowflake check with a static effect.

this would only halve the damage from stun batons, not, say, disablers

**Comment by Rohesie** [comment]
This is not a good direction to go. If you want to add partial resistances, then make them granular and flexible, instead of a binary switch like a trait with an arbitrary amount of soak, such as 50% here.

Use traits to answer boolean questions ("should this mob be affected by stuns?").
